File: server/Ticky.Shared/ViewModels/Auth/TokenResponse.cs
using Ticky.Domain.Entities;

namespace Ticky.Shared.ViewModels.Auth;

public record TokenResponse(
    string AccessToken,
    string RefreshToken,
    DateTime AccessTokenExpiration,
    User User,
    Role[] Roles);

public record User(
    Guid Id,
    string Email,
    IList<string> Roles);

public record Role(
    Guid Id,
    string Name,
    Guid? Extends);

public static class ApplicationRoleExtensions
{
    public static Role ToRole(this ApplicationRole role)
    {
        return new Role(role.Id, role.Name!, role.Extends);
    }
}

File: client/admin-app/src/app/core/models/tokenResponse.ts
import { RoleEnum } from "./role";

export interface TokenResponse {
    accessToken: string;
    refreshToken: string;
    accessTokenExpiration: Date;
    user: User,
    roles: RoleEnum
}

export interface User {
    id: string;
    email: string;
    roles: RoleEnum[]
}

export interface Role {
    id: string;
    name: string;
    extends: string | null
}